kuma/javascript/src/article.jsx

```diff
@@ -89,8 +89,16 @@ function addAnchors(article) {
 
 export default function Article({ document }: DocumentProps) {
     const article = useRef(null);
-    useEffect(() => article.current && highlightSections(article.current));
-    useEffect(() => article.current && addAnchors(article.current));
+    useEffect(() => {
+        if (article.current) {
+            highlightSections(article.current);
+        }
+    }, [document]);
+    useEffect(() => {
+        if (article.current) {
+            addAnchors(article.current);
+        }
+    }, [document]);
```

Comment:
The repetition just looks odd. Is there a strong reason to break up into multiple `useEffect` instead of one?

Reply:
I was wondering the same thing.

Reply:
I'm calling useEffect() twice because these are literally two different effects I'm using. We could combine them, but I think the perf difference would be insignificant.

Note that the two effects were already here in this component. The change here is bug fix where the effects were running too often (and so we were getting multiple link icons in each section heading)